### ISSUE TITLE
Crash in System.Net.WebSockets.Client - Illegal memory access

### ISSUE THREAD

**Issue by karelz** [dotnet/runtime#73621]
8/9:
- 13x hits in [Runfo last 120 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E120+definition%3Aruntime+name%3A%22system.net.websockets.client.tests+work+item%22+targetBranch%3Amain) (older crashes than PR build [1925920](https://runfo.azurewebsites.net/view/build/?number=1925920) on 8/5 are from different reasons)
- 18x hits in Kusto 8/5-8/9 (no crashes failures 6/9-8/4 at all)

Failures are on **Linux only**, with mix of architectures - **arm64, amd64**. Regression started on **8/5**.

2 notable failures are on **Rolling runs**:
- 8/8 AM - [1929089](https://runfo.azurewebsites.net/view/build/?number=1929089)
- 8/9 AM - [1931140](https://runfo.azurewebsites.net/view/build/?number=1931140)

```kusto
let crashes = (friendlyNamePrefix : string, includePR : bool) {
cluster('engsrvprod.kusto.windows.net').database('engineeringdata').WorkItems
    | where FriendlyName startswith friendlyNamePrefix
    //| where Queued > ago(7d)
    | where Status == "BadExit"
    | where ExitCode  == 139
    | join kind=inner (cluster('engsrvprod.kusto.windows.net').database('engineeringdata').Jobs
        | where ((Branch == 'refs/heads/main') or (Branch == 'refs/heads/master') or (includePR and (Source startswith "pr/")) or (Branch startswith 'refs/heads/release/6.0'))
        | where Type startswith "test/functional/cli/"
            and not(Properties contains "runtime-staging")
        | summarize arg_max(Finished, Properties, Type, Branch, Source, Started, QueueName) by JobId
        | project-rename JobType = Type) on JobId
    | extend PropertiesJson = parse_json(Properties)
    | extend OS = replace_regex(tostring(PropertiesJson.operatingSystem), @'\((.*)\).*|([^\(].*)', @'\1\2')
    | extend Runtime = iif(PropertiesJson.runtimeFlavor == "mono", "Mono", iif(PropertiesJson.DefinitionName contains "coreclr", "CoreCLR", ""))
    | extend TargetBranch = extractjson("$.['System.PullRequest.TargetBranch']", Properties)
    | extend Architecture = PropertiesJson.architecture
    | extend Scenario = iif(isempty(PropertiesJson.scenario), "--", PropertiesJson.scenario)
    | project-away PropertiesJson
    | order by Finished desc
};
crashes('System.Net.WebSockets.Client', true);
```

**Comment by CarnaViire** [comment]
https://github.com/dotnet/runtime/pull/73222 was merged on 8/5 so it might be related. The product change in the PR is minimal, but it added a bunch of tests.

cc @greenEkatherine

**Comment by jkotas** [comment]
This is likely the same root cause as #69125. See my comment at https://github.com/dotnet/runtime/issues/69125#issuecomment-1207077374 . You can confirm it by looking at the stacktrace of the crash using unmanaged debugger (lldb, gdb or windbg).

I would expect that any test that depend on sockets is going to hit #69125 with some frequency.

**Comment by wfurt** [comment]
I'm looking into it...

**Comment by hoyosjs** [comment]
Seen in https://dev.azure.com/dnceng/public/_build/results?buildId=1941374&view=results, but that stack is the same as #69125

**Comment by karelz** [comment]
Yes, we expect those 2 are dupes -- it was pending verification from @wfurt

**Comment by wfurt** [comment]
There is no failure in `main` since https://github.com/dotnet/runtime/pull/73972 was merged.

**Comment by karelz** [comment]
Fixed in 8.0 (main) in PR #73972 and in 7.0 (RC1) in PR #74367.
